tabular/src/autogluon/tabular/learner/default_learner.py

```diff
@@ -187,6 +192,17 @@ def general_data_processing(self, X: DataFrame, X_val: DataFrame, X_unlabeled: D
             if X_unlabeled is not None:
                 X_unlabeled = X_super.tail(len(X_unlabeled)).set_index(X_unlabeled.index)
             del X_super
+        if w is not None:
+            X[self.sample_weight] = w
+            if X_val is not None:
+                if w_val is not None:
+                    X_val[self.sample_weight] = w_val
+                elif not self.weight_evaluation:
+                    nan_vals = np.empty((len(X_val),))
+                    nan_vals[:] = np.nan
+                    X_val[self.sample_weight] = nan_vals
```

Comment:
This seems possibly dangerous. Are we sure we want to keep it / have NaNs?

I'd recommend at least adding a TODO here of potentially removing sample_weight column from X_val

Reply:
added TODO, this is necessary as part of the current approach where weights are bundled inside X, but can be removed is we later unbundle